lib/segment/src/index/field_index/numeric_index/tests.rs

```diff
@@ -209,40 +246,27 @@ fn test_payload_blocks_small(#[case] immutable: bool) {
         vec![2.0],
     ];
 
-    values
-        .into_iter()
-        .enumerate()
-        .for_each(|(idx, values)| match &mut index {
-            NumericIndexInner::Mutable(index) => index
-                .add_many_to_list(idx as PointOffsetType + 1, values)
-                .unwrap(),
-            NumericIndexInner::Immutable(_) => unreachable!("index is mutable"),
-        });
-
-    index.flusher()().unwrap();
-
-    // if immutable, we have to reload the index
-    let index = if immutable {
-        let db_ref = index.get_db_wrapper().get_database();
-        let mut new_index: NumericIndexInner<f64> =
-            NumericIndexInner::new(db_ref, COLUMN_NAME, false);
-        new_index.load().unwrap();
-        new_index
-    } else {
-        index
-    };
+    values.into_iter().enumerate().for_each(|(idx, values)| {
+        let values = values.iter().map(|v| Value::from(*v)).collect_vec();
+        let values = values.iter().collect_vec();
+        index_builder
+            .add_point(idx as PointOffsetType + 1, &values)
+            .unwrap();
+    });
+    let index = index_builder.finalize().unwrap();
```

Comment:
Maybe it would be more convenient/concise to create a helper like this and use it instead of calling `add_point()`/`finalize()` in each test.

```rust
pub trait FieldIndexBuilderTrait {
    …
    #[cfg(test)]
    fn make_from_iter(mut self, it: impl Iterator<…>) -> OperationResult<Self::FieldIndexType> { … }
}
```

Reply:
We may do it if other indices require such helper too